samples/oneapi/dpc/ccl/sources/basic_statistics_distr_ccl.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <CL/sycl.hpp>
+#include <iomanip>
+#include <iostream>
+
+#ifndef ONEDAL_DATA_PARALLEL
+#define ONEDAL_DATA_PARALLEL
+#endif
+
+#include "oneapi/dal/algo/basic_statistics.hpp"
+#include "oneapi/dal/io/csv.hpp"
+#include "oneapi/dal/spmd/ccl/communicator.hpp"
+
+#include "utils.hpp"
+
+namespace dal = oneapi::dal;
+
+void run(sycl::queue &queue) {
+    const auto data_file_name = get_data_path("data/covcormoments_dense.csv");
+
+    const auto data =
+        dal::read<dal::table>(queue, dal::csv::data_source{data_file_name});
+
+    const auto bs_desc = dal::basic_statistics::descriptor{};
+
+    auto comm =
+        dal::preview::spmd::make_communicator<dal::preview::spmd::backend::ccl>(
+            queue);
+    auto rank_id = comm.get_rank();
+    auto rank_count = comm.get_rank_count();
+
+    auto input_vec = split_table_by_rows<float>(queue, data, rank_count);
+
+    const auto result = dal::preview::compute(comm, bs_desc, input_vec[rank_id]);
+    if (comm.get_rank() == 0) {
```

Comment:
```suggestion
    if (rank_id == 0) {
```

Reply:
changed all conditions to if (comm.get_rank() == 0) { to get it in one style